File: Facultate/Anul 2/Sem 2/MPP/Laburi/Lab5/JavaTema3/src/main/java/services/AuthService.java
package services;

import models.Employee;
import repos.EmployeeDBRepo;

import java.util.Optional;

public class AuthService {
    private final EmployeeDBRepo employeeDBRepo;

    public AuthService(EmployeeDBRepo employeeDBRepo) {
        this.employeeDBRepo = employeeDBRepo;
    }

    public boolean authenticate(String username, String password){
        if(username.isEmpty() || password.isEmpty()){
            return false;
        }
        Optional<Employee> employee = employeeDBRepo.findByUsername(username);
        return employee.map(value -> value.getPassword().equals(password)).orElse(false);
    }
}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/CSharpTema3/CSharpLab3/services/AuthService.cs
using CSharpLab3.repos;

namespace CSharpLab3.services;

public class AuthService
{
    private readonly IEmployeeRepo _employeeRepo;

    public AuthService(IEmployeeRepo employeeRepo)
    {
        _employeeRepo = employeeRepo;
    }

    public bool Authenticate(string username, string password)
    {
        if (string.IsNullOrEmpty(username) || string.IsNullOrEmpty(password))
            return false;

        var employee = _employeeRepo.FindByUsername(username);
        return employee?.Password == password;
    }
}